prometheus_client/registry.py

```diff
@@ -106,6 +115,25 @@ def collect(self):
 
         return RestrictedRegistry()
 
+    def set_target_info(self, labels):
+        with self._lock:
+            if labels:
+                if not self._target_info and 'target_info' in self._names_to_collectors:
+                    raise ValueError('CollectorRegistry already contains a target_info metric')
+                self._names_to_collectors['target_info'] = None
```

Comment:
I'm a little confused about this one, from my rudimentary understanding (or extreme rudimentary understanding should I say) - every value in `_names_to_collectors` should be a collector yeah?

If None is a value in `_names_to_collectors` and then it gets added to `collectors` var in `restricted_registry`:
```python
        names = set(names)
        collectors = set()
        metrics = []
        with self._lock:
            for name in names:
                if name in self._names_to_collectors:
                    collectors.add(self._names_to_collectors[name])
            if 'target_info' in names and self._target_info:
                metrics.append(self._target_info_metric())
        for collector in collectors:
            for metric in collector.collect():
                samples = [s for s in metric.samples if s[0] in names]
                if samples:
                    m = Metric(metric.name, metric.documentation, metric.type)
                    m.samples = samples
                    metrics.append(m)
```

Wouldn't the `collector.collect()` be a call to `None` from the line `for metric in collector.collect()` if `names` happens to contain `target_info`?